pcap4j-core/src/main/java/org/pcap4j/core/PcapDumper.java

```diff
@@ -203,10 +194,6 @@ public void close() {
 
     dumperLock.writeLock().lock();
     try {
-      if (!open) {
-        logger.warn("Already closed.");
-        return;
-      }
       open = false;
     } finally {
       dumperLock.writeLock().unlock();
```

Comment:
All of above open checks you deleted are needed because the open can be changed by another thread after the previous check.
Please recover them.